pydoctor/sphinx_ext/build_apidocs.py

```diff
@@ -57,6 +53,21 @@ def on_build_finished(app: Sphinx, exception: Exception) -> None:
         # We have a single pydoctor call
         runs = {'main': runs}
 
+    # Defer resolving the git reference for only when asked by
+    # end users.
+    is_git_needed = False
+    for key, value in runs.items():
+        for arg in value:
+            if '{git_reference}' in arg:
+                is_git_needed = True
+                break
```

Comment:
I will apply this if you want. Is just a matter of style.

I prefer to avoid one liners, as they are harder to read. You write once, but the read the code many times... and oftern you read it when you are tired of trying to hunt down a bug.

Some might argue that functional programming is better ... I argue that easy to read (even for a junior) programming is better :)

Reply:
I put it on one line since it was short enough, but it's fine to spread it over multiple lines if you think that makes it more readable.

The important part is `any()`, since that expresses what we're actually trying to compute: is there any argument containing `{git_reference}`?

Reply:
I went with your suggestion as it's not that hard to read..and maybe even easier to read :) Thanks!